### ISSUE TITLE
Upgrade for current Cycle?

### ISSUE THREAD

**Issue by kylecordes** [cyclejs-community/cyclejs-modal#6]
It looks like various Cycle changes have left `cyclejs-modal` behind. I get errors like this trying to compile an app that uses `cyclejs-modal`:

```
[at-loader] ./node_modules/cyclejs-modal/node_modules/@cycle/dom/lib/MainDOMSource.d.ts:6:22 
    TS2420: Class 'MainDOMSource' incorrectly implements interface 'DOMSource'.
  Types of property 'select' are incompatible.
    Type '(selector: string) => DOMSource' is not assignable to type '<S extends DOMSource>(selector: string) => S'.
      Type 'DOMSource' is not assignable to type 'S'. 
```

With a hint in the right direction I could attempt a PR; or if someone is actively maintaining `cyclejs-modal` a keep-up-with-core-Cycle release would be appreciated.

**Comment by jvanbruegge** [comment]
updating the dependencies (ie `@cycle/dom` should be enough)

**Comment by kylecordes** [comment]
Okay, #7. The Cycle part was trivial, but there was an interesting diversion to the vagaries of semver and `@types/____`.

**Comment by jvanbruegge** [comment]
Published in 2.1.0
